src/compile/compile.ts

```diff
@@ -67,6 +65,9 @@ function getTopLevelProperties(topLevelSpec: TopLevel<any>, config: Config) {
 function assembleTopLevelModel(model: Model, topLevelProperties: TopLevelProperties) {
   // TODO: change type to become VgSpec
 
+  // optimize the datafow once
+  optimizeDataflow(model.component.data);
```

Comment:
Why here, not in the `compile` method above?  This feels like a separate phase between `parse` and `assemble`?